sql/sem/builtins: TestSerialNormalizationWithUniqueUnorderedID failed

sql/sem/builtins.TestSerialNormalizationWithUniqueUnorderedID [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4608354&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4608354&tab=artifacts#/) on release-22.1 @ [d7252106b166f23d791a4d677afdbc46b6b059ae](https://github.com/cockroachdb/cockroach/commits/d7252106b166f23d791a4d677afdbc46b6b059ae):


```
=== RUN   TestSerialNormalizationWithUniqueUnorderedID
    builtins_test.go:130: Key counts in each split range
    builtins_test.go:132: range 0: 25
    builtins_test.go:132: range 1: 12
    builtins_test.go:132: range 2: 12
    builtins_test.go:132: range 3: 5
    builtins_test.go:132: range 4: 19
    builtins_test.go:132: range 5: 22
    builtins_test.go:132: range 6: 3
    assertion_compare.go:332: 
        	Error Trace:	builtins_test.go:146
        	Error:      	"30" is not less than "19.5114"
        	Test:       	TestSerialNormalizationWithUniqueUnorderedID
        	Messages:   	[chiSquared value of %f must be less than criticalVal %f to guarantee distribution is relatively uniform 30 19.5114]
--- FAIL: TestSerialNormalizationWithUniqueUnorderedID (1.92s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
/cc @cockroachdb/sql-experience
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSerialNormalizationWithUniqueUnorderedID.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-13953

sql/sem/builtins.TestSerialNormalizationWithUniqueUnorderedID [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4716872&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4716872&tab=artifacts#/) on release-22.1 @ [2d9c7c0869392b757778ac87acc09b7678e9a41d](https://github.com/cockroachdb/cockroach/commits/2d9c7c0869392b757778ac87acc09b7678e9a41d):


```
=== RUN   TestSerialNormalizationWithUniqueUnorderedID
    builtins_test.go:130: Key counts in each split range
    builtins_test.go:132: range 0: 8
    builtins_test.go:132: range 1: 3
    builtins_test.go:132: range 2: 21
    builtins_test.go:132: range 3: 9
    builtins_test.go:132: range 4: 18
    builtins_test.go:132: range 5: 2
    builtins_test.go:132: range 6: 22
    assertion_compare.go:332: 
        	Error Trace:	builtins_test.go:146
        	Error:      	"35.66265060240964" is not less than "19.5114"
        	Test:       	TestSerialNormalizationWithUniqueUnorderedID
        	Messages:   	[chiSquared value of %f must be less than criticalVal %f to guarantee distribution is relatively uniform 35.66265060240964 19.5114]
--- FAIL: TestSerialNormalizationWithUniqueUnorderedID (1.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78411 sql/sem/builtins: TestSerialNormalizationWithUniqueUnorderedID failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSerialNormalizationWithUniqueUnorderedID.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

sql/sem/builtins.TestSerialNormalizationWithUniqueUnorderedID [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=5445781&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=5445781&tab=artifacts#/) on release-22.1 @ [3181b7faa2f3b41d6a15ab4b74d2c60bcfe5132d](https://github.com/cockroachdb/cockroach/commits/3181b7faa2f3b41d6a15ab4b74d2c60bcfe5132d):


```
=== RUN   TestSerialNormalizationWithUniqueUnorderedID
    builtins_test.go:130: Key counts in each split range
    builtins_test.go:132: range 0: 19
    builtins_test.go:132: range 1: 6
    builtins_test.go:132: range 2: 21
    builtins_test.go:132: range 3: 3
    builtins_test.go:132: range 4: 21
    builtins_test.go:132: range 5: 15
    builtins_test.go:132: range 6: 10
    assertion_compare.go:332: 
        	Error Trace:	builtins_test.go:146
        	Error:      	"23.852631578947367" is not less than "19.5114"
        	Test:       	TestSerialNormalizationWithUniqueUnorderedID
        	Messages:   	[chiSquared value of %f must be less than criticalVal %f to guarantee distribution is relatively uniform 23.852631578947367 19.5114]
--- FAIL: TestSerialNormalizationWithUniqueUnorderedID (2.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #78411 sql/sem/builtins: TestSerialNormalizationWithUniqueUnorderedID failed [C-test-failure O-robot T-sql-experience branch-master]
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSerialNormalizationWithUniqueUnorderedID.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/sem/builtins/BUILD.bazel
+++ b/pkg/sql/sem/builtins/BUILD.bazel
@@ -161,6 +161,7 @@ go_test(
         "//pkg/sql/types",
         "//pkg/testutils",
         "//pkg/testutils/serverutils",
+        "//pkg/testutils/skip",
         "//pkg/testutils/sqlutils",
         "//pkg/testutils/testcluster",
         "//pkg/util",

--- a/pkg/sql/sem/builtins/builtins_test.go
+++ b/pkg/sql/sem/builtins/builtins_test.go
@@ -26,8 +26,8 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/sql/sem/tree"
 	"github.com/cockroachdb/cockroach/pkg/sql/types"
 	"github.com/cockroachdb/cockroach/pkg/testutils/serverutils"
+	"github.com/cockroachdb/cockroach/pkg/testutils/skip"
 	"github.com/cockroachdb/cockroach/pkg/testutils/sqlutils"
-	"github.com/cockroachdb/cockroach/pkg/util"
 	"github.com/cockroachdb/cockroach/pkg/util/duration"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
@@ -94,6 +94,8 @@ func TestMapToUniqueUnorderedID(t *testing.T) {
 func TestSerialNormalizationWithUniqueUnorderedID(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	ctx := context.Background()
+	skip.UnderRace(t, "the test is too slow and the goodness of fit test "+
+		"assumes large N")
 	params := base.TestServerArgs{}
 	s, db, _ := serverutils.StartServer(t, params)
 	defer s.Stopper().Stop(ctx)
@@ -110,10 +112,6 @@ CREATE TABLE t (
 )`)
 
 	numberOfRows := 10000
-	if util.RaceEnabled {
-		// We use a small number of rows because inserting rows under race is slow.
-		numberOfRows = 100
-	}
 
 	// Enforce 3 bits worth of range splits in the high order to collect range
 	// statistics after row insertions.
@@ -138,13 +136,13 @@ INSERT INTO t(j) SELECT * FROM generate_series(1, %d);
 	// chi-square goodness of fit statistic. We'll set our null hypothesis as
 	// 'each range in the distribution should have the same probability of getting
 	// a row inserted' and we'll check if we can reject the null hypothesis if
-	// chi-square is greater than the critical value we currently set as 19.5114,
+	// chi-square is greater than the critical value we currently set as 35.2585,
 	// a deliberate choice that gives us a p-value of 0.00001 according to
 	// https://www.fourmilab.ch/rpkp/experiments/analysis/chiCalc.html. If we are
 	// able to reject the null hypothesis, then the distribution is not uniform,
-	// and we raise an error.
+	// and we raise an error. This test has 7 degrees of freedom (categories - 1).
 	chiSquared := discreteUniformChiSquared(keyCounts)
-	criticalValue := 19.5114
+	criticalValue := 35.2585
 	require.Lessf(t, chiSquared, criticalValue, "chiSquared value of %f must be"+
 		" less than criticalVal %f to guarantee distribution is relatively uniform",
 		chiSquared, criticalValue)

